src/components/Button/src/index.tsx

```diff
@@ -32,19 +31,47 @@ export interface ButtonProps extends BaseProps {
   /**
    * Button variant
    */
-  variant?: 'outlined' | 'contained' | 'text';
+  variant?: 'primary-action' | 'secondary-action';
 
   /**
    * HTML type prop
    */
   type?: 'button' | 'submit' | 'reset';
+
+  /**
+   * Icon placed at the start of the button
+   */
+  startIcon?: React.ReactNode;
+
+  /**
+   * Icon placed at the end of the button
+   */
+  endIcon?: React.ReactNode;
 }
 
 /**
  * Primary UI component for user interaction
  */
 export const Button: React.FC<ButtonProps> = (props: ButtonProps) => {
-  return <MaterialButton {...props}>{props.children}</MaterialButton>;
+  const sizeClass = `denhaag-button--${props.size ?? 'medium'}`;
+  let muiVariant: ButtonTypeMap['props']['variant'] = 'contained';
+
+  switch (props.variant) {
+    case 'primary-action':
+      muiVariant = 'contained';
+      break;
+    case 'secondary-action':
+      muiVariant = 'outlined';
+      break;
+  }
+
+  return (
+    <StylesProvider injectFirst>
+      <MaterialButton {...props} classes={classes} className={sizeClass} variant={muiVariant} disableRipple>
```

Comment:
We still pass all props directly into the Material button. Now this is fine for projects using typescript, but what happens when a user uses our components in a default JS react project? Will his application compile when attributes other than the ones specified in our typing are provided? Because if so we are still simply exposing all possible MaterialUI button props and in that case we should only supply the props which are desirable to us to the MaterialButton.

Reply:
Agreed! Avoid committing to a larger API surface than strictly necessary. Removing things is so much harder than adding things, and results in grumpy developers that were depending on those APIs.

Reply:
I don't see how this exposes all possible props of the underlying MaterialUI component? `{...props}` applies every given prop specified in the Properties to the component, no more no less. I am unsure of what happens, if anything, when these components are rendered in JS instead of TS, but I cannot imagine it would fundamentally change the way parameters and/or properties are supplied to a component?

Reply:
> I don't see how this exposes all possible props of the underlying MaterialUI component? `{...props}` applies every given prop specified in the Properties to the component, no more no less. I am unsure of what happens, if anything, when these components are rendered in JS instead of TS, but I cannot imagine it would fundamentally change the way parameters and/or properties are supplied to a component?

I think Maarten was referring to the fact that js wouldn't have type safety, and therefore a user could provide a material UI prop which we don't define in our interface, which would then be passed on to the underlying Material component. I'll be removing the {...props}